pkg/datagatherer/k8s/dynamic.go

```diff
@@ -371,6 +375,43 @@ func redactList(list []*api.GatheredResource) error {
 
 			// remove managedFields from all resources
 			Redact(RedactFields, resource)
+
+			annotsRaw, ok, err := unstructured.NestedFieldNoCopy(resource.Object, "metadata", "annotations")
+			if err != nil {
+				return fmt.Errorf("wasn't able to find the metadata.annotations field: %w", err)
+			}
```

Comment:
I don't think this should return an error, because the object might not have any annotations. I guess it depends whether the `metadata.annotations` field has an `omitempty` marker.

Reply:
You are right. I will rework this and add more unit tests to feel a little more confident.